splice_ck/src/main/java/com/splicemachine/ck/HBaseInspector.java

```diff
@@ -0,0 +1,176 @@
+                                    tableId.set(new Pair<>(userData.getTimestamp(), er.getColumn(SYSTABLESRowFactory.SYSTABLES_TABLEID).getString()));
+                                }
+                            }
+                        }
+                    };
+                    for (Cell cell : r.listCells()) {
+                        cellVisitor.visit(cell);
+                    }
+                }
+                if (tableId.get() == null) {
+                    throw new TableNotFoundException();
+                }
+                return constructSchemaFromSysColsTable(tableId.get().getSecond());
+            }
+        }
+    }
+
+    private Utils.Tabular constructSchemaFromSysColsTable(String tableId) throws Exception {
+        String systableId = getHbaseRegionOf(SYSCOLUMNSRowFactory.TABLENAME_STRING);
+        try (Connection conn = ConnectionFactory.createConnection(config)) {
+            Table table = conn.getTable(TableName.valueOf(systableId));
+            Scan scan = new Scan().addColumn(SIConstants.DEFAULT_FAMILY_BYTES, SIConstants.PACKED_COLUMN_BYTES);
+            try (ResultScanner results = table.getScanner(scan)) {
+                final ValContainer<Utils.Tabular> schema = new ValContainer<>(new Utils.Tabular("column index", "column name", "column type"));
+                for (Result r : results) {
+                    ICellVisitor cellVisitor = new ICellVisitor() {
+                        @Override
+                        protected void visitUserData(Cell userData) throws StandardException {
+                            UserDataDecoder decoder = new SysColsDataDecoder();
+                            ExecRow er = decoder.decode(userData);
+                            if (er.getColumn(SYSCOLUMNSRowFactory.SYSCOLUMNS_REFERENCEID) != null
+                                    && tableId.equals(er.getColumn(SYSCOLUMNSRowFactory.SYSCOLUMNS_REFERENCEID).getString())) {
+                                schema.get().addRow(Integer.toString(er.getColumn(SYSCOLUMNSRowFactory.SYSCOLUMNS_COLUMNNUMBER).getInt()),
+                                        CheckNull(er.getColumn(SYSCOLUMNSRowFactory.SYSCOLUMNS_COLUMNNAME).toString()),
+                                        CheckNull(((TypeDescriptorImpl)(er.getColumn(SYSCOLUMNSRowFactory.SYSCOLUMNS_COLUMNDATATYPE)).getObject()).getTypeName()));
+                            }
+                        }
+                    };
+                    for (Cell cell : r.listCells()) {
+                        cellVisitor.visit(cell);
+                    }
+                }
+                return schema.get();
+            }
+        }
+    }
+
+    public String getHbaseRegionOf(String tableName) throws Exception {
+        Utils.Tabular results = getListTables();
+        Utils.Tabular.Row r = results.rows.stream().filter(result -> result.cols.get(2 /*name*/).equals(tableName)
```

Comment:
tableName

Reply:
done. I use instead of index number a properly named integer constant.

```
public String regionOf(String tableName) throws Exception {
        Utils.Tabular results = getListTables();
        Utils.Tabular.Row row = results.rows.stream().filter(result -> result.cols.get(TBL_TABLES_NAME_IDX).equals(tableName)
                && result.cols.get(TBL_TABLES_INDEX_IDX).equals(NULL)).min((l, r) -> Long.compare(Long.parseLong(r.cols.get(TBL_TABLES_CREATE_TXN_IDX)),
                Long.parseLong(l.cols.get(TBL_TABLES_CREATE_TXN_IDX)))).orElseThrow(TableNotFoundException::new);
        assert row.cols.size() > 0;
        return row.cols.get(TBL_TABLES_HBASE_NAME_IDX);
    }
```